src/lib/SpeculationEngine.ts

```diff
@@ -14,22 +14,21 @@ export class SpeculationEngine {
     return queued.findIndex(({ id }) => id === landRequestStatus.id);
   }
 
-  static async getImpact(queued: LandRequestStatus[], position: number) {
-    const currentRequestStatus = queued[position];
-    const nextRequestStatus = queued[position + 1];
+  static async getImpact(queued: LandRequestStatus[], position: number, availableSlots: number) {
+    // Compare current request with availableSlots - 1 number of next queue items
+    const queuedRequestsLength = position + availableSlots - 1;
+    const nextQueuedRequestStatus = queued.slice(position, queuedRequestsLength + 1);
 
-    const currentImpact = currentRequestStatus.request.impact;
-    const nextImpact = nextRequestStatus.request.impact;
+    const queuedRequestImpacts = nextQueuedRequestStatus.map((requestStatus) => {
+      return requestStatus.request.impact;
+    });
 
-    Logger.info('Impact retrieved:', {
+    Logger.info('Impact retrieved for next queued requests:', {
       namespace: 'lib:speculationEngine:getImpact',
-      pullRequestId: currentRequestStatus.request.pullRequestId,
-      nextPullRequestId: nextRequestStatus.request.pullRequestId,
-      currentImpact,
-      nextImpact,
+      impact: queuedRequestImpacts,
```

Comment:
can we also add `pullRequestId` to the logging for the current PR. It will be useful for debugging via Splunk.